packages/frontend-v2/pages/signup.tsx

```diff
@@ -64,97 +39,90 @@ const SignUpForm = () => {
       router.push("/home");
     } catch (err) {
       const error = err as AxiosError;
-      if (error.response?.status === 401) toast.error("Invalid Credentials!");
-      else toast.error("Something went wrong!");
+      if (error.response?.data?.message === emailAlreadyExistsError) {
+        toast.error(
+          "Account with the given email already exists... try signing up instead 😄"
+        );
+      } else {
+        handleApiClientError(error, router);
+      }
     }
   };
 
   return (
-    <FormFormat onSubmit={handleSubmit(onSubmitHandler)}>
-      <SignUpFormTopInput
-        errors={errors}
-        password={password}
-        register={register}
-      />
-      <SignUpFormButton isSubmitting={isSubmitting} router={router} />
-    </FormFormat>
+    <AuthForm
+      onSubmit={handleSubmit(onSubmitHandler)}
+      headingText="Create an Account"
+      inputs={
+        <>
+          <Flex columnGap="md">
+            <GraduateInput
+              type="text"
+              id="firstName"
+              placeholder="First Name"
+              {...register("firstName")}
```

Comment:
So this won't work because react hook form will simply create a field `firstName` in the payload and send that to the BE when it should be `fullName` which includes both `firstName` and `lastName`. We need to pull out both of these values on submit and then combine them together into `fullName` because right now, the post request is not setting `fullName` in the BE. 

Another approach we can take is to make `firstName` and `lastName` distinct in the BE, which I think is better because it can be difficult to guess first and last name from just `fullName`. Someone with the name "Yu Ping Chao" would have "Yu Ping" as their first name, but would be difficult to identify if we just stored `fullName: "Yu Ping Chao"` in the DB.

Reply:
ahhh good catch, will update this!